src/ui/public/index_patterns/index_patterns.js

```diff
@@ -41,6 +41,7 @@ export function IndexPatternsProvider(Notifier, Private) {
   self.cache = patternCache;
   self.getIds = getProvider('id');
   self.getTitles = getProvider('attributes.title');
+  self.getIndexPatterns = getProvider();
```

Comment:
++ I like it.  Though I don't know if I like the name `getSavedObjects`.  We use `savedDashboard`, `savedVis` in other places.  Maybe `getSavedIndexPatterns` to match?